# FIFA 19 — Manipulação Básica

## 1. Carregando o dataset

In [ ]:
import pandas as pd

df = pd.read_csv('fifa.csv',index_col=0)
df.head()

,ID,Name,Age,Photo,Nationality,Flag,Overall,Potential,Club,Club Logo,...,Composure,Marking,StandingTackle,SlidingTackle,GKDiving,GKHandling,GKKicking,GKPositioning,GKReflexes,Release Clause
0,158023,L. Messi,31,https://cdn.sofifa.org/players/4/19/158023.png,Argentina,https://cdn.sofifa.org/flags/52.png,94,94,FC Barcelona,https://cdn.sofifa.org/teams/2/light/241.png,...,96.0,33.0,28.0,26.0,6.0,11.0,15.0,14.0,8.0,€226.5M
1,20801,Cristiano Ronaldo,33,https://cdn.sofifa.org/players/4/19/20801.png,Portugal,https://cdn.sofifa.org/flags/38.png,94,94,Juventus,https://cdn.sofifa.org/teams/2/light/45.png,...,95.0,28.0,31.0,23.0,7.0,11.0,15.0,14.0,11.0,€127.1M
2,190871,Neymar Jr,26,https://cdn.sofifa.org/players/4/19/190871.png,Brazil,https://cdn.sofifa.org/flags/54.png,92,93,Paris Saint-Germain,https://cdn.sofifa.org/teams/2/light/73.png,...,94.0,27.0,24.0,33.0,9.0,9.0,15.0,15.0,11.0,€228.1M
3,193080,De Gea,27,https://cdn.sofifa.org/players/4/19/193080.png,Spain,https://cdn.sofifa.org/flags/45.png,91,93,Manchester United,https://cdn.sofifa.org/teams/2/light/11.png,...,68.0,15.0,21.0,13.0,90.0,85.0,87.0,88.0,94.0,€138.6M
4,192985,K. De Bruyne,27,https://cdn.sofifa.org/players/4/19/192985.png,Belgium,https://cdn.sofifa.org/flags/7.png,91,92,Manchester City,https://cdn.sofifa.org/teams/2/light/10.png,...,88.0,68.0,58.0,51.0,15.0,13.0,5.0,10.0,13.0,€196.4M


## 2. Fatiamento

Criando um novo df com as características principais dos jogadores

In [59]:
df = df[['Name','Age','Nationality','Club','Overall','Potential']]
df.head()

,Name,Age,Nationality,Club,Overall,Potential
0,L. Messi,31,Argentina,FC Barcelona,94,94
1,Cristiano Ronaldo,33,Portugal,Juventus,94,94
2,Neymar Jr,26,Brazil,Paris Saint-Germain,92,93
3,De Gea,27,Spain,Manchester United,91,93
4,K. De Bruyne,27,Belgium,Manchester City,91,92


## 3. Filtro Simples

Filtragem simples do país mais vencedor da Copa do Mundo FIFA

In [64]:
filtro_brasil = df.loc[df['Nationality'] == 'Brazil']

In [63]:
filtro_brasil.describe()

,Age,Overall,Potential
count,827.000000,827.000000,827.000000
mean,27.625151,71.251511,73.351874
std,4.243753,5.590515,5.785576
min,17.000000,48.000000,59.000000
25%,25.000000,68.000000,69.000000
50%,27.000000,71.000000,73.000000
75%,30.000000,74.000000,77.000000
max,40.000000,92.000000,93.000000


## 4. Filtro Múltiplo

In [67]:
filtro_barca = df.loc[df['Club'].str.contains('Barcelona', na=False)]

,Name,Age,Nationality,Club,Overall,Potential
0,L. Messi,31,Argentina,FC Barcelona,94,94
7,L. Suárez,31,Uruguay,FC Barcelona,91,91
18,M. ter Stegen,26,Germany,FC Barcelona,89,92
20,Sergio Busquets,29,Spain,FC Barcelona,89,89
32,Coutinho,26,Brazil,FC Barcelona,88,89
42,S. Umtiti,24,France,FC Barcelona,87,92
49,Jordi Alba,29,Spain,FC Barcelona,87,87
53,I. Rakitić,30,Croatia,FC Barcelona,87,87
54,Piqué,31,Spain,FC Barcelona,87,87
96,A. Vidal,31,Chile,FC Barcelona,85,85


In [68]:
filtro_barca.describe()

,Age,Overall,Potential
count,33.000000,33.000000,33.000000
mean,23.848485,78.030303,85.303030
std,4.711165,9.583465,4.837903
min,18.000000,61.000000,75.000000
25%,19.000000,69.000000,82.000000
50%,23.000000,82.000000,86.000000
75%,29.000000,87.000000,89.000000
max,32.000000,94.000000,94.000000


In [70]:
filtro_brasil_over = df.loc[(df['Nationality'] == 'Brazil') & (df['Overall'] >= 85)]
filtro_brasil_over.sort_values('Overall', ascending=False)

,Name,Age,Nationality,Club,Overall,Potential
2,Neymar Jr,26,Brazil,Paris Saint-Germain,92,93
27,Casemiro,26,Brazil,Real Madrid,88,90
32,Coutinho,26,Brazil,FC Barcelona,88,89
35,Marcelo,30,Brazil,Real Madrid,88,88
39,Thiago Silva,33,Brazil,Paris Saint-Germain,88,88
57,Ederson,24,Brazil,Manchester City,86,90
61,Roberto Firmino,26,Brazil,Liverpool,86,87
64,Alex Sandro,27,Brazil,Juventus,86,86
65,Douglas Costa,27,Brazil,Juventus,86,86
75,Fernandinho,33,Brazil,Manchester City,86,86


In [71]:
filtro_brasil_over.describe()

,Age,Overall,Potential
count,15.000000,15.000000,15.000000
mean,28.800000,86.600000,87.533333
std,3.569714,1.919821,2.445599
min,24.000000,85.000000,85.000000
25%,26.000000,85.000000,85.500000
50%,27.000000,86.000000,87.000000
75%,32.500000,88.000000,89.500000
max,35.000000,92.000000,93.000000


In [75]:
filtro_pot = df.loc[df['Potential'] > 90]
filtro_pot.sort_values('Age', ascending=True)

,Name,Age,Nationality,Club,Overall,Potential
1143,Vinícius Júnior,17,Brazil,Real Madrid,77,92
226,M. de Ligt,18,Netherlands,Ajax,82,91
25,K. Mbappé,19,France,Paris Saint-Germain,88,95
229,G. Donnarumma,19,Italy,Milan,82,93
155,O. Dembélé,21,France,FC Barcelona,83,92
156,Gabriel Jesus,21,Brazil,Manchester City,83,92
55,L. Sané,22,Germany,Manchester City,86,92
79,Marco Asensio,22,Spain,Real Madrid,85,92
56,Bernardo Silva,23,Portugal,Manchester City,86,91
177,Kepa,23,Spain,Chelsea,83,91


In [76]:
filtro_pot.describe()

,Age,Overall,Potential
count,29.000000,29.000000,29.000000
mean,24.793103,87.551724,92.034483
std,4.220522,3.978390,1.148998
min,17.000000,77.000000,91.000000
25%,22.000000,85.000000,91.000000
50%,25.000000,88.000000,92.000000
75%,27.000000,91.000000,93.000000
max,33.000000,94.000000,95.000000


## Conclusões

* Temos **827** jogadores brasileiros na database do jogo, com uma média de idade de **27** anos, overall de **71** e **73** de potencial
* O time do Barcelona tem **33** jogadores registrados, com uma média de idade de **23** anos, overall de **78** e **85** de potencial
* Temos **15** jogadores brasileiros com **85+** de overall, tendo uma média de **28** anos, Neymar Jr. do PSG é o melhor jogador do Brasil com **92** de overall.
* Por fim, temos 29 jogadores com Potential acima de 90, sendo **K. Mbappé** com o maior potencial, podendo chegar até **95**, e o mais novo sendo **Vinícius Júnior** que pode chegar a **92** de overall.